install/kubernetes/helm/istio/templates/configmap.yaml

```diff
@@ -24,13 +24,23 @@ data:
     # Deprecated: mixer is using EDS
     {{- if .Values.mixer.enabled }}
     {{- if .Values.global.controlPlaneSecurityEnabled }}
+    {{- if .Values.global.proxy.discoveryDomain }}
+    mixerCheckServer: istio-policy.{{ .Release.Namespace }}.svc.{{ .Values.global.proxy.discoveryDomain }}:15004
+    mixerReportServer: istio-telemetry.{{ .Release.Namespace }}.svc.{{ .Values.global.proxy.discoveryDomain }}:15004
+    {{- else }}
     mixerCheckServer: istio-policy.{{ .Release.Namespace }}.svc.cluster.local:15004
     mixerReportServer: istio-telemetry.{{ .Release.Namespace }}.svc.cluster.local:15004
+    {{- end }}
+    {{- else }}
```

Comment:
This if else end else logic is a little hard to follow.

Can you try (not sure if this will work) to be a little more obvious:
`{{- if ne Values.global.controlPlaneSecurityEnabled }}`